source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -171,33 +167,60 @@ void TextureBaker::writeBakedDocument(ConstShaderRefPtr shaderRef, const FilePat
 
     // Create top-level elements.
     NodeGraphPtr bakedNodeGraph = bakedTextureDoc->addNodeGraph("NG_baked");
-    GeomInfoPtr bakedGeom = (udimSetValue) ? bakedTextureDoc->addGeomInfo("GI_baked") : nullptr;
+    GeomInfoPtr bakedGeom = !udimSet.empty() ? bakedTextureDoc->addGeomInfo("GI_baked") : nullptr;
     if (bakedGeom)
     {
-        bakedGeom->setGeomPropValue("udimset", udimSetValue->getValueString(), "stringarray");
+        bakedGeom->setGeomPropValue("udimset", udimSet, "stringarray");
     }
     MaterialPtr bakedMaterial = bakedTextureDoc->addMaterial("M_baked");
-    ShaderRefPtr bakedShaderRef = bakedMaterial->addShaderRef(shaderRef->getName() + "_baked", shaderRef->getAttribute("node"));
+    ShaderRefPtr bakedShaderRef = bakedMaterial->addShaderRef(_shaderRef->getName() + "_baked", _shaderRef->getAttribute("node"));
     if (_baseType == Image::BaseType::UINT8)
     {
         bakedNodeGraph->setColorSpace("srgb_texture");
     }
 
     // Create bind elements on the baked shader reference.
-    for (ValueElementPtr valueElem : shaderRef->getChildrenOfType<ValueElement>())
+    for (ValueElementPtr valueElem : _shaderRef->getChildrenOfType<ValueElement>())
     {
         BindInputPtr bindInput = valueElem->asA<BindInput>();
         if (bindInput && bindInput->getConnectedOutput())
         {
             OutputPtr output = bindInput->getConnectedOutput();
 
-            // Create the baked bind input.
+            // Create the baked bindinput.
             BindInputPtr bakedBindInput = bakedShaderRef->addBindInput(bindInput->getName(), bindInput->getType());
 
+            // Store a constant value for uniform outputs.
+            if (_uniformOutputs.count(output))
+            {
+                Color4 uniformColor = _bakedImageMap[output][0].uniformColor;
```

Comment:
Could this part be decomposed into a helper function (e.g. setConstantValueString) at the top?

Reply:
Good suggestion, Madeleine, and I'll make that change.